nsc-events-nextjs-15-593-modify-the-styling-of-edit-users-page

bennettsf opened the pull request:
Resolves #593 

This pull request modifies/refactors the styling and overall functionality of the "Edit Users" page where admin accounts can edit a specific user's role. The biggest change is using mui's Table components to closely match the figma design given in sprint 15. Because of this, the UserCard component won't be needed (ignore any changes made to it in this PR).

Before this PR gets accepted, I want to make sure with leadership that it's how we envisioned the page to look. It will still need more features added (search bar, pagination, and other possible user data edits) but these are just my initial test changes.

For testing, make sure your test database is up and running. Try editing the role of a user and then double checking that the change went through to the backend. Here is an example:

![44bbrJF](https://github.com/user-attachments/assets/4e3fd9c0-18ba-40c6-8276-1f5b8a57bd7b)

After making those role changes, you can see that they went through on my test database:

![drKJEcv](https://github.com/user-attachments/assets/8405671a-6291-44d2-9a5b-6d356a9c8bd6)
![0kaOk8z](https://github.com/user-attachments/assets/4fdec3af-f099-4452-91e0-8c904198d618)

bennettsf commented:
The build for this PR is currently failing because we'll need to write new tests for editUserRole.test.tsx with the refactoring. I'll spend some time on that this week.

Asfand00 reviewed (commented):
Hey @bennettsf, I tested your branch, and everything looks good on my end. I like the look of the page so far. As soon as your building issue check gets resolved the PR will get an approve by me (or someone else if I don't check in time). 

For testing I signed in to my admin account and changed the role of my user account to creator (the first account in the image): 
![nsc-events-edit-user-role](https://github.com/user-attachments/assets/960aeafa-e315-403c-88ed-b7a0dc1cb557)
Then double checked in mongoDB:
![nsc-events-edit-user-role2](https://github.com/user-attachments/assets/56bb3b81-338f-4746-a1f8-84b01680d82f)

bennettsf commented:
- Got rid of the old UserCard component and created a UserTable component to be used on the page.

- Modified the tests for the page to fix the build error and properly test the new component render.

- Added a hover effect for each row so the user is aware of what user they are clicking the edit button for.

I think this is a good stopping point for this PR and more features can be added to the page in future PRs.

BradleyCharles reviewed (approved):
Tested and looking good. Good work!
